packages/prop-controllers/src/data.ts

```diff
@@ -13,3 +14,15 @@ export const lengthDataSchema = z.object({
 })
 
 export type LengthData = z.infer<typeof lengthDataSchema>
+
+export const gapDataSchema = z.object({
+  value: z.number(),
+  unit: z.literal('px'),
+})
+
+export type GapData = z.infer<typeof gapDataSchema>
```

Comment:
I would name this `LengthData` and the one above `LengthPercentageData`.

For reference:
- https://developer.mozilla.org/en-US/docs/Web/CSS/length
- https://developer.mozilla.org/en-US/docs/Web/CSS/length-percentage

Basically, `type LengthPercentageData = LengthData | PercentageData`

Although our version of `<length>` doesn't support units other than `px`. We might add that eventually.

Reply:
We already have `LengthData` and it supports percentage:
<img width="410" alt="image" src="https://github.com/makeswift/makeswift/assets/13066728/39b5b25c-7a3b-41a8-abc1-d238713ca76e">

I'll keep this as `GapData` for now.